interview-questions.md

```diff
@@ -11,57 +11,83 @@ Then, research the question to expand on your answer. Even if you feel you have
+
 
 
 4. What is JSX? What is one syntax difference between HTML and JSX?
 
   Your answer:
 
-  Researched answer:
+  JSX is a type of code used in React. It is a mix of HTML and JavaScript, that allows you to intertwine JavaScript within HTML. The syntax is a little bit different when using JSX because when starting an element in HTML you will distinguish the JavaScript inside with {JavaScript} around the portion that calls on JavaScript. Also, when giving a tag an attribute, the camelcase will be incorporated in it, instead of normal HTML. An example of this is <p className=""> vs <p class="">. There are many similarities but in the end the subtle differences make them very different.
 
+  Researched answer:
 
+  JSX is JavaScript flavored HTML syntax. As stated above, JSX uses className vs class because class is solely HTML syntax. The JSX will be rendered and lives inside of fragment tags so that we can give it unlimited nested items for the application.
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
   Your answer:
 
-  Researched answer:
+  Yarn is used through your terminal to run, start, test, or create certain applications. Running React through your terminal with yarn start will allow you to use your local device to act as a network for you to see and interact with your application. It will refresh as soon as the file is saved, and giving that real time interaction allows you to modify and see your application as a user would once deployed. Yarn can also be used to test your code for cleaner code with less bugs. Yarn is also able to create a React App right through your terminal, using yarn create react-app. Files that are modified in a React application when you run yarn are those that are coded under the source folder or any of the given folders that are tied to React with "create react-app" with yarn as well.
 
+  Researched answer:
 
+  Yarn is the package manager. The files that are all nested within the package or included when you create or start React with Yarn, will be brought to your code editor. The files that are modified are the ones you have imported to your main app.js file.
 
 6. STRETCH: What is an anonymous function in JavaScript?
 
   Your answer:
 
+  An anonymous function is a function that hasn't been given a name but it is still a valid function. An example could be something like:
+  var hi = function () {
+    console.log('hello')
+    }
+
   Researched answer:
 
+  Most commonly an anonymous function is used as an argument to other functions. It is only accessible by the variable it stored in. An anonymous function can have multiply arguments. Giving it multiple arguments is a pro for developers to use anonymous functions in this way.
```

Comment:
Anonymous functions are not stored to variables, as a result, they can only be called on only in the instance they are used.